ios/sdk/WeexSDK/Sources/Engine/WXSDKEngine.m

```diff
@@ -72,6 +72,8 @@ + (void)_registerDefaultComponents
     [self registerComponent:@"image" withClass:NSClassFromString(@"WXImageComponent") withProperties:nil];
     [self registerComponent:@"scroller" withClass:NSClassFromString(@"WXScrollerComponent") withProperties:nil];
     [self registerComponent:@"list" withClass:NSClassFromString(@"WXListComponent") withProperties:nil];
+    [self registerComponent:@"recycler" withClass:NSClassFromString(@"WXRecyclerComponent") withProperties:nil];
+    [self registerComponent:@"waterfall" withClass:NSClassFromString(@"WXRecyclerComponent") withProperties:nil];
```

Comment:
YES, waterfall is more understandable to use for multi-column layout, and recycler will be a fundamental component to be used for multiple layout, such as linear/flex.